pulser-core/pulser/backend/results.py

```diff
@@ -202,6 +202,12 @@ def from_abstract_repr(cls, repr: str) -> Results:
         d = json.loads(repr)
         return cls._from_abstract_repr(d)
 
+    def __getstate__(self) -> dict:
+        return self._to_abstract_repr()  # type: ignore[operator]
+
+    def __setstate__(self, d: dict) -> None:
+        self.__dict__ = d
```

Comment:
I would like to see this checked...
I have seen more `self.__dict__.update(d)`
Otherwise I think you could go for
```suggestion
    from dataclasses import fields
    def __setstate__(self, d: dict) -> None:
        new_result = self.from_abstract_repr(json.dumps(d, cls=AbstractReprEncoder)) # that way, you have some validation of your incoming dict, which is useful if the class changes for instance
        for f in fields(new_result):  # Using fields is more in the spirit of a dataclass 
             setattr(self, f.name, getattr(new_result, f.name))
```